### PR TITLE
Fix AddComponent(target) not working

### LINKED ISSUE #39122: COMPONENT_DUPE_HIGHLANDER doesn't work
Reproduction steps:
1. get a chair(/obj/structure/chair)
2. pick up the chair and plant it(use in hand) multiple times
3. view variables of chair
4. search for the datum_components variable
5. notice that there are multiple of the component(/datum/component/simple_rotation) that is added to the chair in ComponentInitialize()

I noticed this when creating a component that I would use TransferComponents() a lot with, since the component created on ComponentInitialize() doesn't get removed.

### PR DESCRIPTION
**Pull request by Garen7**
[Changelogs]: # (Your PR should contain a detailed changelog of notable changes, titled and categorized appropriately. This includes, new features, sprites, sounds, balance changes, admin tools, map edits, removals, big refactors, config changes, hosting changes and important fixes. An example changelog has been provided below for you to edit. If you need additional help, read https://github.com/tgstation/tgstation/wiki/Changelogs)

:cl: Garen
fix: Fixed AddComponent(target) not working when an instance is passed rather than a path
/:cl:

[why]: # (Please add a short description [two lines down] of why you think these changes would benefit the game. If you can't justify it in words, it might not be worth adding.)

Fixes: #39122

### CODE REVIEW

File: code/datums/components/_component.dm [hunk 1]
```diff
@@ -237,7 +237,10 @@
 
 	if(dm != COMPONENT_DUPE_ALLOWED)
 		if(!dt)
-			old_comp = GetExactComponent(nt)
+			if(ispath(nt))
+				old_comp = GetExactComponent(nt)
+			else
+				old_comp = GetExactComponent(nt.type)
```
Comment (ninjanomnom) [inline code/datums/components/_component.dm]: Prety sure we shouldnt be doing addcomponent with instances anymore. You want `target.TakeComponent(comp_instance)` instead

**Review by ninjanomnom — CHANGES_REQUESTED** [review]

File: code/datums/components/_component.dm (hunk 1 above)
Comment (optimumtact) [reply code/datums/components/_component.dm]: @Garen7

Comment (Garen7) [reply code/datums/components/_component.dm]: Sorry I was busy I’m going to fix it tomorrow

Comment (ninjanomnom) [reply code/datums/components/_component.dm]: I double checked this and I was mistaken earlier. However do `nt = new_comp.type` in the else of the ispath check above instead.

**Review by ninjanomnom — APPROVED** [review]

**Pull request merged**